### ISSUE TITLE
baremetal: Use pinned RHCOS version for masters

### ISSUE THREAD

**Issue by russellb** [openshift/installer#2064]
Original issue from @markmc here: https://github.com/openshift-metal3/kni-installer/issues/37

---

openshift-install is currently built with a pinned build of rhcos. See e.g.

```
hack/build.sh:5:RHCOS_BUILD_NAME="${RHCOS_BUILD_NAME:-400.7.20190306.0}"
```

We currently properly use this image for the bootstrap VM launched with the libvirt provider:

```
        case baremetal.Name:
                osimage, err = rhcos.QEMU(ctx, rhcos.DefaultChannel)
```

This resolves to a URL in that channel for the QEMU image for that build as listed in meta.json at https://releases-rhcos.svc.ci.openshift.org/storage/releases

However, for the masters, we currently require that you pass that in via the following in install-config.yaml:

```
    master_configuration:
      image_source: "http://172.22.0.1/images/rhcos-maipo-latest.qcow2"
      image_checksum: 7656f23da8caf42142c0eafde22db096
```

and we generate this image in dev-scripts by taking the openstack qcow image and making a "dualdhcp" variant.

We should be working towards using an rhcos image for masters that is published in the official channels, and resolving the URL for that from published metadata like we do for the libvirt qcow.

**Comment by russellb** [comment]
@markmc - May 7

---

The end goal here should probably be that bare-metal IPI fully manages the provisioning service for both master and worker deployment, including providing the correct RHCOS images to the provisioning service

i.e. openshift-install should download the RHCOS image and give it to Ironic. And longer-term when the a new release payload can update the bootimage in the installer-created MachineSets, in the bare-metal case it should download the new RHCOS image and give it to Ironic

**Comment by russellb** [comment]
@hardys noted on the original issue that this should be resolved by #2060

**Comment by hardys** [comment]
> @hardys noted on the original issue that this should be resolved by #2060

Yes this should be resolved as part of the rework for #2037 which is a prerequisite for #2060 - I'll take this issue and work on a PR which switches to using the image maintained internally vs the install-config data.

@hardys noted on the original issue that this should be resolved by #2060

**Comment by juliakreger** [comment]
@dhellmann FYI
